spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/TraceIdLoggingWebMvcInterceptor.java

```diff
@@ -35,22 +34,28 @@
  * trace ID metadata to log messages.
  *
  * @author Mike Eltsufin
+ * @author Chengyuan Zhao
  */
 public class TraceIdLoggingWebMvcInterceptor extends HandlerInterceptorAdapter {
 
-	private static final String X_CLOUD_TRACE = "x-cloud-trace-context";
+	private final TraceIdExtractor traceIdExtractor;
 
-	private static final String X_B3_TRACE = "X-B3-TraceId";
+	public TraceIdLoggingWebMvcInterceptor(TraceIdExtractor extractor) {
+		Assert.notNull(extractor, "A valid trace id extractor is required.");
+		this.traceIdExtractor = extractor;
+	}
 
-	private static final List<String> TRACE_HEADERS = ImmutableList.of(X_CLOUD_TRACE,
-			X_B3_TRACE);
+	public TraceIdExtractor getTraceIdExtractor() {
+		return this.traceIdExtractor;
+	}
 
 	@Override
 	public boolean preHandle(HttpServletRequest req,
 			HttpServletResponse resp, Object handler) throws Exception {
-		String traceId = extractTraceIdFromRequest(req);
+		String traceId = this.traceIdExtractor.extractTraceIdFromRequest(req);
 		if (traceId != null) {
-			TraceLoggingEnhancer.setCurrentTraceId(extractTraceIdFromRequest(req));
+			TraceLoggingEnhancer.setCurrentTraceId(
+					this.traceIdExtractor.extractTraceIdFromRequest(req));
```

Comment:
You have just extracted `traceId` local variable. A fresh `extractTraceIdFromRequest()` call may return a different value.
I'm sure you've been going to use that `traceId` local variable here as well